protos/NUgus.proto

```diff
@@ -120,11 +159,32 @@ PROTO NUgus [
                 children [
                   Shape {
                     appearance PBRAppearance {
+
+                      %{
+                        if fields.name.value ~= '' then
+                          -- name is supposed to be something like "red player 2" or "blue player 1"
```

Comment:
You shouldn't need to create these variables again, the `color` and `number` variables still exist from before from what I understand of how this works.